internal/provider/resource_gitlab_project_issue_board.go

```diff
@@ -0,0 +1,219 @@
+package provider
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"strconv"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	gitlab "github.com/xanzy/go-gitlab"
+)
+
+var _ = registerResource("gitlab_project_issue_board", func() *schema.Resource {
+	return &schema.Resource{
+		Description: `The ` + "`" + `gitlab_project_issue_board` + "`" + ` resource allows to manage the lifecycle of a Project Issue Board.
+
+~> **NOTE:** If the board lists are changed all lists will be recreated.
+
+**Upstream API**: [GitLab REST API docs](https://docs.gitlab.com/ee/api/boards.html)`,
+
+		CreateContext: resourceGitlabProjectIssueBoardCreate,
+		ReadContext:   resourceGitlabProjectIssueBoardRead,
+		UpdateContext: resourceGitlabProjectIssueBoardUpdate,
+		DeleteContext: resourceGitlabProjectIssueBoardDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Schema: gitlabProjectIssueBoardSchema(),
+	}
+})
+
+func resourceGitlabProjectIssueBoardCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*gitlab.Client)
+
+	project := d.Get("project").(string)
+	options := gitlab.CreateIssueBoardOptions{
+		Name: gitlab.String(d.Get("name").(string)),
+	}
+
+	log.Printf("[DEBUG] create Project Issue Board %q in project %q", *options.Name, project)
+	issueBoard, _, err := client.Boards.CreateIssueBoard(project, &options, gitlab.WithContext(ctx))
```

Comment:
@timofurrer  - Since this is a "Create then update" style resource, it feels like we should run "setId" before we run the update in the event that the "create" succeeds but the subsequent update fails. While board names are not required to be unique (so a second run using "create" wouldn't necessarily error), it would result in a unmanaged board being successfully created for every run if we don't update the state before a failed update, right?

Reply:
@PatrickRice-KSC great catch! Just updated the PR 🎉